specifyweb/frontend/js_src/lib/components/Formatters/Definitions.tsx

```diff
@@ -0,0 +1,246 @@
+  readonly table: SpecifyTable;
+}): JSX.Element {
+  const isReadOnly = React.useContext(ReadOnlyContext);
+  const [isExpanded, _, __, handleToggle] = useBooleanState(false);
+  return (
+    <div
+      className={`flex ${
+        isExpanded || !hasCondition ? 'flex-col' : 'items-center'
+      } gap-2`}
+      key={index}
+    >
+      {hasCondition && (
+        <Label.Block>
+          {isExpanded ? resourcesText.conditionFieldValue() : null}
+          <div className="flex items-center gap-2">
+            <div className="flex-1">
+              <Input.Text
+                className="h-full"
+                isReadOnly={isReadOnly}
+                value={value ?? ''}
+                onValueChange={(value): void =>
+                  handleChanged(
+                    {
+                      value: value.length === 0 ? undefined : value,
+                      fields,
+                    },
+                    index
+                  )
+                }
+              />
+            </div>
+            <span className="-ml-2" />
+            {trimmedFieldsLength > 0 && isExpanded ? (
+              <Button.Danger
+                onClick={(): void => {
+                  handleChange(removeItem(formatter.definition.fields, index));
+                }}
+              >
+                {resourcesText.deleteDefinition()}
+              </Button.Danger>
+            ) : null}
+            {hasCondition && isExpanded ? (
+              <Button.Icon
+                icon={isExpanded ? 'chevronUp' : 'chevronDown'}
+                title="showConditionalField"
+                onClick={handleToggle}
+              />
+            ) : null}
+          </div>
+          {isExpanded ? (
+            <span>
+              {index === 0
+                ? resourcesText.elseConditionDescription()
+                : resourcesText.conditionDescription()}
+            </span>
+          ) : null}
+        </Label.Block>
+      )}
+      {isExpanded || !hasCondition ? null : fields.length === 0 ? (
+        <Button.Small
+          onClick={(): void => {
+            handleToggle();
+            handleChanged(
+              {
+                value,
+                fields: [
+                  {
+                    separator: localized(' '),
+                    aggregator: undefined,
+                    formatter: undefined,
+                    fieldFormatter: undefined,
+                    field: undefined,
+                  },
+                ],
+              },
+              index
+            );
+          }}
+        >
+          {resourcesText.addField()}
+        </Button.Small>
+      ) : (
+        fields.map((field, index) => (
+          <p key={index}>
+            {field.separator === undefined ? '' : field.separator}
+            {field.field === undefined ? '' : field.field[0].label}
+          </p>
+        ))
+      )}
+      {isExpanded || !hasCondition ? (
```

Comment:
you are using `isExpanded || !hasCondition` and `!isExpanded && hasCondition` a lot of times in this component - can you refactor by creating a variable for it and then using that variable and !variable?